### PR TITLE
Design story workspace service

### PR DESCRIPTION
**Pull request by rounak610**
### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update

### Checklist
- [ ] My pull request is atomic and focuses on a single change.
- [ ] I have documented my changes clearly and comprehensively.
- [ ] I have added the required tests.

### CODE REVIEW

File: workspace-service/app/services/impl/docker_workspace_service.go [hunk 1]
```diff
@@ -190,6 +207,11 @@ func (ws DockerWorkspaceService) checkAndCreateWorkspaceFromTemplate(workspaceId
 		ws.logger.Error("Failed to chown workspace", zap.Error(err))
 		return err
 	}
+	err = utils.ChownRWorkspace("1000", "1000", frontendPath)
```
Comment (czezi) [inline workspace-service/app/services/impl/docker_workspace_service.go:210]: This is not need as frontend is already a part of workspace path.

File: workspace-service/app/services/impl/k8s_workspace_service.go [hunk 2]
```diff
@@ -323,6 +341,11 @@ func (ws K8sWorkspaceService) checkAndCreateWorkspaceFromTemplate(workspaceId st
 		ws.logger.Error("Failed to chown workspace", zap.Error(err))
 		return err
 	}
+	err = utils.ChownRWorkspace("1000", "1000", frontendPath)
```
Comment (czezi) [inline workspace-service/app/services/impl/k8s_workspace_service.go:344]: Not needed

**Review by czezi — CHANGES_REQUESTED** [review]

File: workspace-service/app/controllers/workspace_controller.go [hunk 3]
```diff
@@ -21,7 +21,7 @@ func (wc *WorkspaceController) CreateWorkspace(c *gin.Context) {
 		})
 		return
 	}
-	wsDetails, err := wc.wsService.CreateWorkspace(body.WorkspaceId, *body.BackendTemplate, body.RemoteURL, body.GitnessUserName, body.GitnessToken)
+	wsDetails, err := wc.wsService.CreateWorkspace(body.WorkspaceId, *body.BackendTemplate,  *body.FrontendTemplate, body.RemoteURL, body.GitnessUserName, body.GitnessToken)
```
Comment (czezi) [inline workspace-service/app/controllers/workspace_controller.go:24]: Keep the frontend template as string pointer. We need to consider it optional.

File: workspace-service/app/services/impl/docker_workspace_service.go [hunk 4]
```diff
@@ -76,11 +79,25 @@ func (ws DockerWorkspaceService) checkAndCreateWorkspaceFromTemplate(workspaceId
...
 	ws.logger.Info("Creating workspace from template", zap.String("workspaceId", workspaceId), zap.String("backendTemplate", backendTemplate))
 
+	//copying backend template in root dir
 	err = utils.SudoRsyncFolders("/templates/"+backendTemplate+"/", "/workspaces/"+workspaceId)
 	if err != nil {
 		ws.logger.Error("Failed to rsync folders", zap.Error(err))
 		return err
 	}
+	//creating a frontend folder in the directory
+	frontendPath := "/workspaces/"+workspaceId+"/frontend"
```
Comment (czezi) [inline workspace-service/app/services/impl/docker_workspace_service.go:89]: Check if it's optional and skip creation.

**Review by czezi — CHANGES_REQUESTED** [review]

File: workspace-service/app/services/impl/docker_workspace_service.go (hunk 4 above)
Comment (rounak610) [reply workspace-service/app/services/impl/docker_workspace_service.go]: made the frontendTemplate optional, only if it not nil then only the frontend template will be copied

**Pull request merged**